src/core/server/opensearch/client/client_config.ts

```diff
@@ -120,6 +121,10 @@ export function parseClientOptions(config: OpenSearchClientConfig, scoped: boole
     clientOptions.disablePrototypePoisoningProtection = config.disablePrototypePoisoningProtection;
   }
 
+  if (config.compression) {
+    clientOptions.compression = config.compression;
```

Comment:
do we want to add a test for this given we updated the test to not do this by default?